src/Features/LanguageServer/Protocol/Workspaces/LspWorkspaceManager.cs

```diff
@@ -93,202 +83,58 @@ internal class LspWorkspaceManager : IDocumentChangeTracker, IDisposable
         _requestTelemetryLogger = requestTelemetryLogger;
 
         _lspWorkspaceRegistrationService = lspWorkspaceRegistrationService;
-        // This server may have been started after workspaces were registered, we need to ensure we know about them.
-        foreach (var workspace in lspWorkspaceRegistrationService.GetAllRegistrations())
-        {
-            OnWorkspaceRegistered(this, new LspWorkspaceRegisteredEventArgs(workspace));
-        }
-
-        lspWorkspaceRegistrationService.WorkspaceRegistered += OnWorkspaceRegistered;
     }
 
     public void Dispose()
     {
-        // Workspace events can come in while we're disposing of an LSP server (e.g. restart).
-        lock (_gate)
-        {
-            _lspWorkspaceRegistrationService.WorkspaceRegistered -= OnWorkspaceRegistered;
-            foreach (var registeredWorkspace in _workspaceToLspSolution.Keys)
-            {
-                registeredWorkspace.WorkspaceChanged -= OnWorkspaceChanged;
-            }
-
-            _workspaceToLspSolution.Clear();
-        }
-    }
-
-    #region Workspace Updates
-
-    /// <summary>
-    /// Handles cases where a new workspace is registered after an LSP server is already started.
-    /// For example, in VS the metadata as source workspace is not created until a metadata files is opened.
-    /// </summary>
-    private void OnWorkspaceRegistered(object? sender, LspWorkspaceRegisteredEventArgs e)
-    {
-        lock (_gate)
-        {
-            // Set the LSP solution for the workspace to null so that when asked we fork from the workspace.
-            _workspaceToLspSolution.Add(e.Workspace, null);
-            e.Workspace.WorkspaceChanged += OnWorkspaceChanged;
-
-            _logger.TraceInformation($"Registered workspace {e.Workspace.Kind}");
-        }
+        _cachedLspSolutions.Clear();
```

Comment:
not really needed.  if we're being disposed, presumably we'll be GC'ed a moment later, and this clears anyways.  could remove this and even remove IDisposable from this type